### ISSUE TITLE
if actions are sent in an entity, MART says "forgot action parameter"

### ISSUE THREAD

**Issue by mdutoo** [cgourdin/MartServer#46]
If actions are sent in an entity, MART return an HTTP 400 error with "you cannot use an action with PUT method":

```
{
  "id": "urn:uuid:d99486b7-0632-482d-a184-a9195733ddd3",
  "title": "compute4",
  "summary": "My only compute for test with single resource",
  "kind": "http://schemas.ogf.org/occi/infrastructure#compute",
  "mixins": [],
  "attributes": {
    "occi.core.id": "urn:uuid:d99486b7-0632-482d-a184-a9195733ddd3",
    "occi.compute.architecture": "x86",
    "occi.compute.cores": 4,
    "occi.compute.hostname": "test",
    "occi.compute.share": 0,
    "occi.compute.speed": 3,
    "occi.compute.memory": 4,
    "occi.compute.state": "inactive"
  },
  "actions": [
    "http://schemas.ogf.org/occi/infrastructure/compute/action#start",
    "http://schemas.ogf.org/occi/infrastructure/compute/action#stop",
    "http://schemas.ogf.org/occi/infrastructure/compute/action#restart",
    "http://schemas.ogf.org/occi/infrastructure/compute/action#suspend",
    "http://schemas.ogf.org/occi/infrastructure/compute/action#save"
  ],
  "location": "compute/d99486b7-0632-482d-a184-a9195733ddd3"
}
```

However, if actions JSON is not sent within, the entity resource is updated rightly.

Doesn't OCCI forbid sending actions (which are a categories / kind thing) in entities ? What is their meaning ?

**Comment by cgourdin** [comment]
If i understand you need to put a resource with actions scheme + term  (full entity definition including actions of its kinds), MART use the "actions" part when parsing the query but as PUT is invoked MART says "you can't do that".
I dont think OCCI forbid sending actions as this is a part of resource definition. I don't use actions definitions in my tests because there is no meaning for me to include them in a PUT (to avoid payload) and in your example this is a part of a kind definition (compute actions here). 
In a PUT request this part must be ignored. To end this, this is a bug of MART, it must be ignored on PUT.

Action invocation is made by POST requests :
The json rendering doc give that : 
<pre>
<code>
{
”action ”: String ,
”attributes ”: Object
 }
</code>
</pre>

**Comment by mdutoo** [comment]
OK. I've found that sample somewhere, I don't remember where, it's not mine.

**Comment by cgourdin** [comment]
ok, but i will ignore it in put request.

**Comment by mdutoo** [comment]
OK. To be clear, OCCInterface needs the action array to be accepted by the server without doing anything with it, because the easiest way for OCCInterface to change resources is to modify an existing one and send it back again.

NB. This is caused by #51 REST semantics.

**Comment by cgourdin** [comment]
Also render now an entity response on json accept media type with PUT method.

**Comment by cgourdin** [comment]
Has been fixed in 18cc0a0
